src/iv/ivgl.cpp

```diff
@@ -903,6 +937,44 @@ IvGL::paint_pixelview()
 
 
 
+void
+IvGL::paint_datawindow()
+{
+    IvImage* img = m_current_image;
+    const ImageSpec& spec(img->spec());
+
+    glDisable(GL_TEXTURE_2D);
+    glUseProgram(0);
+
+    glColor4f(0.5f, 0.5f, 0.5f, 1.f);
+    const float xmin = spec.x;
+    const float xmax = spec.x + spec.width;
+    const float ymin = spec.y;
+    const float ymax = spec.y + spec.height;
+    gl_rect_border(xmin, ymin, xmax, ymax);
```

Comment:
I'm not so sure this is correct. `(spec.x, spec.y)` is the upper-left pixel. But `(spec.x + spec.width, spec.y + spec.height)` is ONE PAST the lower right pixel in each direction.

So first of all, I think we should either draw the rectangle RIGHT ON the boundary pixels, or SURROUNDING the boundary pixels, but definitely not one choice on one side and the other choice on the other side. 

I'm also not sure which is more helpful. Right on actually obscures the top and bottom row and left and right columns. Maybe it should be just beyond the boundary?

Or am I misunderstanding the whole issue?

Reply:
If lines were rasterized at pixel center, what you said would be true (with last row and column being covered). But in this case, the bound `(spec.x, spec.y)`, `(spec.x + spec.width, spec.y + spec.height)` lie on the pixel edges. It's uncertain which side the fragments are going to be rasterized. And the actual line segment rasterization algorithm isn't strictly enforced by the OpenGL specification, so it can differ from device to device. This results in the following cases, with the same zoom level but panning the camera slightly to the right yields different rasterized lines.
|Border shown on top and left|Border shown on top and right|
|:--:|:--:|
|<img width="316" alt="image" src="https://github.com/user-attachments/assets/78a28199-0156-432e-a3dd-9f061c09617d">|<img width="318" alt="image" src="https://github.com/user-attachments/assets/4f96e281-8f98-46dd-860a-08bc1d1e1eea">|

I don't see this as a major concern, as modern displays have high dpi standards. It should be quite trivial for users to notice.

Reply:
OK, let's roll with this oddity for now. We can always revisit later if it's causing actual trouble for people.